File: ea/eaBase.py
import math

class eaBase:
 ''' class with functions 
   to calculate ea from 
   mean relative humidity and
   minimum and maximum measurements.
   This formula is not very accurate
   but is used when no RHmin and RHmax
   are available
   '''

  
 def e(self,t):
  '''method that
   calculates the e0
   from temperature'''
   
  klasma=17.27*t/(t+237.3)
  return 0.6108*math.exp(klasma)

 def calc_ea(self):
  return None

 def calc_es(self,tmin,tmax):
  '''This method will calculate the es using the above
  methods
  '''
  return (self.e(tmin)+self.e(tmax))/2

if __name__ == '__main__':
  eabase=eaBase()
  print eabase.calc_es(15,24.6)

File: ocaml/eaBase.ml
class eaBase = object (self)
 (*
  2class with functions
   to calculate ea from
   mean relative humidity and
   minimum and maximum measurements.
   This formula is not very accurate
   but is used when no RHmin and RHmax
   are available
  *) 
  method e t =
    (*
     *method that
     *calculates the e0
     *from temperature
     *)
    0.6108*.exp(17.27*.t/.(t+.237.3))
  
  method calc_es tmin tmax = 
    (*
     * This method will calculate the es using the above 
     * methods
    *)
    (self#e tmin +. self#e tmax )/.2.0

end;;

let () = 
  (* create an object*)
  let obj = new eaBase  in
  Printf.printf "%f\n" (obj#calc_es 15.0 24.6)
